lib/checkuninitvar.cpp

```diff
@@ -1653,6 +1653,8 @@ void CheckUninitVar::valueFlowUninit()
                 if (!v->subexpressions.empty() && usage == ExprUsage::PassedByReference)
                     continue;
                 if (usage != ExprUsage::Used) {
+                    if (tok->isUnaryOp("&") && !tok->astParent())
```

Comment:
This check should probably be added to the `getExprUsage` function so it returns `ExprUsage::NotUsed`.